src/cns.ts

```diff
@@ -123,6 +124,16 @@ export default class Cns extends EthereumNamingService {
     return addr;
   }
 
+  /**
+   * Produces CNS namehash
+   * @param domain - domain to be hashed
+   * @returns CNS namehash of a domain
+   */
+  namehash(domain: string): string {
+    this.ensureSupportedDomain(domain);
+    return hash(domain);
```

Comment:
Please add tests on the domains being properly namehashed when they start or end with `-`.